Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -519,77 +519,92 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
                     
-                    // Update the cache with the resolved node.
-                    // We aggressively release used memory, since we're copying all semantic objects
-                    // on the line below while rewriting nodes with the resolved content.
-                    documentationNode.semantic = autoreleasepool { resolver.visit(documentationNode.semantic) }
+            let inheritanceParentReference: ResolvedTopicReference?
+            // Check if this documentation is inherited.
+            if let symbolSemantic = documentationNode.semantic as? Symbol,
+               let origin = symbolSemantic.origin,
+               let originReference = documentationCache.reference(symbolID: origin.identifier)
+            {
+                inheritanceParentReference = originReference
+                
+                // Check if we should skip resolving links for inherited documentation
+                if !externalMetadata.inheritDocs,
+                   // Check that this symbol only has documentation from an in-source documentation comment
+                   documentationNode.docChunks.count == 1,
+                   case .sourceCode = documentationNode.docChunks.first?.source,
+                   // Check that that documentation comment is inherited from a symbol belonging to another module
+                   let originSymbolSemantic = documentationCache[originReference]?.semantic as? Symbol,
+                   symbolSemantic.moduleReference != originSymbolSemantic.moduleReference
+                {
+                    // Don't resolve any links for this symbol.
+                    return
+                }
+            } else {
+                inheritanceParentReference = nil
+            }
+            
+            var resolver = ReferenceResolver(context: self, bundle: bundle, rootReference: reference, inheritanceParentReference: inheritanceParentReference)
+            
+            // Update the node with the markup that contains resolved references instead of authored links.
+            documentationNode.semantic = autoreleasepool { 
+                // We use an autorelease pool to release used memory as soon as possible, since the resolver will copy each semantic value
+                // to rewrite it and replace the authored links with resolved reference strings instead.
+                resolver.visit(documentationNode.semantic)
+            }
+            
+            let problems: [Problem]
+            if documentationNode.semantic is Article {
+                // Diagnostics for articles have correct source ranges and don't need to be modified.
+                problems = resolver.problems
+            } else {
+                // Diagnostics for in-source documentation comments need to be offset based on the start location of the comment in the source file.
+                
+                // Get the source location
+                var inSourceDocumentationCommentSource: URL?
```

Comment:
Similarly, you could move this loop into a private function inside of `DocumentationNode` and then have a couple of computed properties, `inSourceCommentSource` and `inSourceCommentOffset` or even shorted `inSourceUrl` and `inSourceOffset`.

Reply:
Extracted in https://github.com/apple/swift-docc/pull/951/commits/baa0539a3119bab76a9e12d7adbe88afdb303895